contracts/src/TokenLib.sol

```diff
@@ -6,178 +6,48 @@ pragma solidity 0.8.28;
 
 import {IERC20} from "./interfaces/IERC20.sol";
 import {IERC20Permit} from "./interfaces/IERC20Permit.sol";
+import {ECDSA} from "openzeppelin/utils/cryptography/ECDSA.sol";
 
 library TokenLib {
-    // keccak256('EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)')
-    bytes32 internal constant DOMAIN_TYPE_SIGNATURE_HASH =
-        bytes32(0x8b73c3c69bb8fe3d512ecc4cf759cc79239f7b179b0ffacaa9a75d522b39400f);
-
-    // keccak256('Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)')
-    bytes32 internal constant PERMIT_SIGNATURE_HASH =
-        bytes32(0x6e71edae12b1b97f4d1f60370fef10105fa2faae0126114a169c64845d6126c9);
-
-    string internal constant EIP191_PREFIX_FOR_EIP712_STRUCTURED_DATA = "\x19\x01";
-
     struct Token {
         mapping(address account => uint256) balance;
         mapping(address account => mapping(address spender => uint256)) allowance;
         mapping(address token => uint256) nonces;
         uint256 totalSupply;
     }
 
-    /**
-     * @dev See {IERC20-transfer}.
-     *
-     * Requirements:
-     *
-     * - `recipient` cannot be the zero address.
-     * - the caller must have a balance of at least `amount`.
-     */
-    function transfer(Token storage token, address sender, address recipient, uint256 amount) external returns (bool) {
-        _transfer(token, sender, recipient, amount);
-        return true;
-    }
-
-    /**
-     * @dev Creates `amount` tokens and assigns them to `account`, increasing
-     * the total supply.
-     *
-     * Emits a {Transfer} event with `from` set to the zero address.
-     *
-     * Requirements:
-     *
-     * - `to` cannot be the zero address.
-     */
     function mint(Token storage token, address account, uint256 amount) external {
-        if (account == address(0)) {
-            revert IERC20.InvalidAccount();
-        }
-
+        require(account != address(0), IERC20.InvalidReceiver(address(0)));
```

Comment:
Why not use `revert` as it’s used in other cases?

Reply:
`require` does revert with the custom error `IERC20.InvalidReceiver` if the condition is not met.

Reply:
Can `amount` be 0 here?

Reply:
Yes, it seems most ERC20 implementations allow 0-value transfers, to allow for better composability with DeFI apps.

More convo here: https://x.com/PaulRBerg/status/1644318310165626880

Reply:
Added a test in https://github.com/Snowfork/snowbridge/pull/1384/commits/254cb361fe198777a5ea674a49dadbf955329148 to ensure zero-value transfers are allowed.